### ISSUE TITLE
Add support for SSH private keys with a passphrase when running a BBS migration

### ISSUE THREAD

**Issue by timrogers** [github/gh-gei#747]
When performing a migration from Bitbucket Server, an SSH username and private key are required for connecting to the server and downloading the migration archive.

If your SSH private key is protected with a passphrase, an error is triggered and it is not possible to continue with your migration:

```
[16:26] [ERROR] Renci.SshNet.Common.SshPassPhraseNullOrEmptyException: Private key is encrypted but passphrase is empty.
```

We should add support for using private keys with a passphrase.

**Comment by robandpdx** [comment]
Can't users load their key into ssh-agent? Does this solution not work?

**Comment by timrogers** [comment]
I don't believe that using an SSH agent is supported - my SSH agent didn't kick in - but that would, in my view, be a great way to fulfil the intent of this issue.
